pytube/__main__.py

```diff
@@ -120,20 +110,25 @@ def descramble(self) -> None:
             self.player_config_args = self.vid_info
         else:
             assert self.watch_html is not None
-            self.player_config_args = get_ytplayer_config(self.watch_html)[
-                "args"
-            ]
+            self.player_config_args = get_ytplayer_config(self.watch_html)["args"]
 
             # Fix for KeyError: 'title' issue #434
-            if "title" not in self.player_config_args:  # type: ignore
-                i_start = self.watch_html.lower().index("<title>") + len(
-                    "<title>"
-                )
-                i_end = self.watch_html.lower().index("</title>")
-                title = self.watch_html[i_start:i_end].strip()
-                index = title.lower().rfind(" - youtube")
+            # ---> ADD THIS PART <---
+            if 'title' not in self.player_config_args:
+                # for more reliability when parsing, we may use a trained parser
+                try:
+                    from bs4 import BeautifulSoup
```

Comment:
I think for only extracting the title, a html parser is way too resource heavy. It should be able to be done with a regex only.